typing/oprint.ml

```diff
@@ -459,38 +459,65 @@ let out_sig_item = ref (fun _ -> failwith "Oprint.out_sig_item")
 let out_signature = ref (fun _ -> failwith "Oprint.out_signature")
 let out_type_extension = ref (fun _ -> failwith "Oprint.out_type_extension")
 
-let rec print_out_functor funct ppf =
-  function
-    Omty_functor (_, None, mty_res) ->
-      if funct then fprintf ppf "() %a" (print_out_functor true) mty_res
-      else fprintf ppf "functor@ () %a" (print_out_functor true) mty_res
-  | Omty_functor (name, Some mty_arg, mty_res) -> begin
-      match name, funct with
-      | "_", true ->
-          fprintf ppf "->@ %a ->@ %a"
-            print_simple_out_module_type mty_arg
-            (print_out_functor false) mty_res
-      | "_", false ->
+let rec print_out_module_type ppf mty =
+  print_out_functor ppf mty
```

Comment:
`print_out_module_type` is split in several precedence levels, first functors then simple module types. `print_out_functor` thus does not mean that we know that we have a functor, it means that we are at the precedence level of functors or below.